src/wallet/wallet.cpp

```diff
@@ -511,12 +511,29 @@ void CWallet::SetBestChain(const CBlockLocator& loc)
 std::set<std::pair<libzcash::PaymentAddress, uint256>> CWallet::GetNullifiersForAddresses(const std::set<libzcash::PaymentAddress> & addresses)
 {
     std::set<std::pair<libzcash::PaymentAddress, uint256>> nullifierSet;
+    // Sapling ivk -> addr map
+    std::map<libzcash::SaplingIncomingViewingKey, libzcash::SaplingPaymentAddress> ivkMap;
+    for (const auto & addr : addresses) {
+      if (boost::get<libzcash::SaplingPaymentAddress>(&addr) != nullptr) {
+        auto saplingAddr = boost::get<libzcash::SaplingPaymentAddress>(addr);
+        libzcash::SaplingIncomingViewingKey ivk;
+        this->GetSaplingIncomingViewingKey(saplingAddr, ivk);
+        ivkMap.insert(std::make_pair(ivk, saplingAddr));
```

Comment:
I think that one ivk can have multiple sapling addresses. If this is the case then we will end up returning more nullifiers than we need. I think that the way we use this method (to determine if a note is change), it would not be wrong to return all nullifiers for the wallet, just unnecessary. It's worth it to double check this.